File: java praktikum/prak12/src/prak12/Soal3.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package prak12;

import java.util.Scanner;

/**
 *
 * @author Nrhys
 */
public class Soal3 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int[] data = new int[20];
        int i = 0;
        
        while(i < data.length){
            System.out.println("inputkan data ke-"+(i+1)+": ");
            data[i] = input.nextInt();
            i++;
        }
        
        System.out.println("Setelah dicari = ");
        hitung(data);
        
    }
    
    static void hitung(int[] data){
        int i = 1, jumlahsekarang = 0, panjangsekarang = 1, panjangterpanjang = 0, indeksterpanjang = 0, indekssekarang = 0, countlarge = 0;
        
        while(i < 20){
            if(data[i] >= data[i-1]){
                jumlahsekarang += data[i-1];
                panjangsekarang += 1;
                
            }else {
                jumlahsekarang += data[i-1];
                if(jumlahsekarang > countlarge){
                    countlarge = jumlahsekarang;
                    panjangterpanjang = panjangsekarang;
                    indeksterpanjang = indekssekarang;
                }else{
                    
                }
                jumlahsekarang = 0;
                panjangsekarang = 1;
                indekssekarang = i;
                
            }
            i++;
        }
        
        i = indeksterpanjang;
        while(i <= indeksterpanjang + panjangterpanjang - 1){
            System.out.print(data[i] + ", ");
            i++;
        }
        System.out.println("(" + countlarge +")");
    }
    
    
    
}

File: Python praktikum/prak5/soal3.py
data = [0] * (100000)

i = 0
while i < 20:
    data[i] = int(input("Inputkan data ke-" + str(i+1) + " : "))
    i = i + 1
data[20] = 0
i = 1
penjumlahansekarang = 0
panjangsekarang = 1
panjangterpanjang = 0
indeksterpanjang = 0
indekssekarang = 0
penjumlahanterpangjang = 0
while i <= 20:
    if data[i] >= data[i - 1]:
        penjumlahansekarang = penjumlahansekarang + data[i - 1]
        panjangsekarang = panjangsekarang + 1
        i = i + 1
    else:
        penjumlahansekarang = penjumlahansekarang + data[i - 1]
        if penjumlahansekarang > penjumlahanterpangjang:
            penjumlahanterpangjang = penjumlahansekarang
            panjangterpanjang = panjangsekarang
            indeksterpanjang = indekssekarang
        penjumlahansekarang = 0
        panjangsekarang = 1
        indekssekarang = i
        i = i + 1
for i in range(indeksterpanjang, indeksterpanjang + panjangterpanjang - 1 + 1, 1):
    print(data[i])
print("( " + str(penjumlahanterpangjang) + " )")
